### DISCUSSION TITLE
RotaryHandleSensor don't change color on "open"

### DISCUSSION THREAD

**Discussion by Herbert-Testmann** [SukramJ/aiohomematic#1831] [Ideas]
Hallo

Fenster Drehgriff Sensoren HM-SEC-RHS und HMIP-SRH wechseln auf dem Dashboard nicht die Symbolfarbe, wenn der Zustand ungleich "geschlossen" ist.

Das betrifft offenbar devicetyps class RotaryHandleSensor und class RotaryHandleSensorIP

Die Geräte können den Status 0 oder 1 oder 2 haben.

Warum ändert das Symbol auf dem Dashboard nicht die Farbe, wenn der Status ungleich "0" ist?
Das Symbol ändert sich, nicht aber die Farbe.

Kann ich dazu ein Issues auf machen, oder gibt es Gründe, warum das nicht geht oder nicht gewollt ist ?

  
<img width="390" alt="Bildschirmfoto 2024-11-10 um 11 32 22" src="https://github.com/user-attachments/assets/da74d893-5558-4013-92ff-ef63a42afa91">
<img width="398" alt="Bildschirmfoto 2024-11-10 um 12 04 37" src="https://github.com/user-attachments/assets/32385fdc-ed6b-4ef2-9a5e-1b0803d54091">

**Comment by SukramJ** [comment]
Diese Sensoren haben 3 Zuständen und  nicht 2, wie ein `binary_sensor`, für die es in HA eine entsprechende `device_class` gibt.

Ansonsten gibt es vielleicht [hier](https://github.com/danielperna84/hahomematic/wiki/Integration-of-three-state-window-handle-sensors-HmIP-SRH#integration-of-three-state-window-handle-sensors-hmip-srh-by-lateadapter) was nützliches.

**Comment by Herbert-Testmann** [comment]
Vielen Dank für den Link zur Anleitung.
Das Einbinden von User Icons habe ich nicht hin bekommen. Die Integration installiert und die 3 Icon Dateien in einen neuen Ordner, wie in der Anleitung beschrieben. Die Icons sollen bei allen Entitäten zur Auswahl stehen, sie sind aber nicht zu sehen.
Ist auch nicht wichtig, weil ich mich für "case 4" entschieden habe. Mir reicht die Änderung der Farbe bei Redutierung auf 2 Zustände.

Dazu habe ich meine vorhandene customize.yaml mit folgendem Eintrag ergänzen wollen. Die Syntax stimmt aber nicht :(

```
- binary_sensor:
  - name: "eg_ez_f2"
    device_class: "window"
    state: >
        {% if is_state('sensor.eg_ez_f2_1', 'open') %}
            on
        {% elif is_state('sensor.eg_ez_f2_1', 'closed') %}
            off
        {% else %}
            on
        {% endif %}

   
```     


Die gesamte Datei sieht dann so aus.  Ist das die falsche Stelle für die Anpassung oder einfach "nur" Syntax?


```
sensor.ccu3_webui_sv_w_counter_1:
  state_class: total_increasing
  unit_of_measurement: "kWh"
  device_class: energy
  name: w_counter_1
  friendly_name: Strom Zählerstand
  
- binary_sensor:
  - name: "eg_ez_f2"
    device_class: "window"
    state: >
        {% if is_state('sensor.eg_ez_f2_1', 'open') %}
            on
        {% elif is_state('sensor.eg_ez_f2_1', 'closed') %}
            off
        {% else %}
            on
        {% endif %}

```
